hw02-diy-arraylist/src/main/java/my/homeworks/hw02/DiyArrayList/DiyArrayList.java

```diff
@@ -0,0 +1,156 @@
+package my.homeworks.hw02.DiyArrayList;
+
+import java.util.*;
+
+public class DiyArrayList<T> implements List<T>
+{
+    /**
+     * Коэффициент расширения хранилища
+     */
+    final int kExtend = 2;
```

Comment:
```suggestion
    private static final int K_EXTEND = 2;
```

Лучше `private` и `static`, тк константа.
И константы принято называть большими буквами через подчеркивание.